File: 2015/10.py
"""--- Day 10: Elves Look, Elves Say ---"""

def iterate(string):
  new = []
  count = 0
  compare = string[0]
  for s in string:
    if s == compare:
      count += 1
    else:
      new.append(str(count) + compare)
      # reset
      count = 1
      compare = s
  new.append(str(count) + string[-1])
  return ''.join(new)

with open("input/10.txt") as instructions:
  for line in instructions:
    text = str(line).replace("\n", "")

  for i in range(1, 51):
    text = iterate(text)
    if i in (40, 50):
      print(len(text))

File: 2015/10.go
package main

import (
	"fmt"
)

func lookAndSay(input []byte) []byte {
	var new []byte
	count := byte('0')
	compare := input[0]

	for _, v := range input {
		if v == compare {
			count++
		} else {
			new = append(new, count, compare)
			// reset
			count = byte('1')
			compare = v
		}
	}
	return append(new, count, input[len(input)-1])
}

func main() {
	res := lookAndSay([]byte("1113222113"))

	for i := 1; i < 50; i++ {
		if i == 40 {
			fmt.Println(len(res))
		}
		res = lookAndSay(res)
	}
	fmt.Println(len(res))
}
